.github/workflows/backend.yml

```diff
@@ -0,0 +1,83 @@
+name: KFP Tekton backend unit tests
+
+on:
+  push:
+    branches: [master]
+
+  # Run tests for any PRs which change the backend code
+  pull_request:
+    paths:
+      - 'go.mod'
+      - 'backend/**'
+      - 'scripts/deploy/github/**'
+      - 'manifests/kustomize/**'
+
+env:
+  GITHUB_ACTION: "true"
+  SETUPTOOLS_USE_DISTUTILS: "stdlib"
+
+jobs:
+  run-go-unittests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: 1.20.x
+    - name: Checkout code
+      uses: actions/checkout@v4
+    - name: "run go unit tests"
+      run: go test -v -cover ./backend/...
+  backend-integration:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Create k8s Kind Cluster
+        uses: container-tools/kind-action@v2
+        with:
+          cluster_name: kfp-tekton
+          kubectl_version: v1.26.3
```

Comment:
I found the deployment is missing some manifests file. It probably due to the recent update to the new kustomize 5.3 version. To fix that, we need to upgrade the kubectl version to 1.32+ or convert all the kubectl -k commands to `kustomize`

Reply:
I changed to 1.29.4 because of the k8s 1.29 support we're planning for Kubeflow 1.9. I'll check the kubectl -k change to kustomize.